File: app/schemas/payroll.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class PayrollOut(BaseModel):
    id: int
    user_id: int
    month: str
    base_salary: int
    bonus_amount: int
    penalty_amount: int
    total_paid: int
    paid_at: Optional[datetime]
    comment: Optional[str]

    class Config:
        orm_mode = True


class PayrollUpdate(BaseModel):
    base_salary: Optional[int] = None
    bonus_amount: Optional[int] = None
    penalty_amount: Optional[int] = None
    comment: Optional[str] = None

File: frontend/src/types/payroll.ts
export interface Payroll {
  id: number;
  user_id: number;
  month: string;
  base_salary: number;
  bonus_amount?: number;
  penalty_amount?: number;
  kpi_percent?: number;
  kpi_rule_id?: number;
  total_paid: number;
  comment?: string;
  paid_at?: string;
  created_at: string;
  updated_at?: string; 
  user?: {
    id: number;
    full_name: string;
    email?: string;
    position?: {
      id: number;
      name: string;
    };
    role?: {
      id: number;
      name: string;
    };
  };
}

export interface PayrollCreate {
  user_id: number;
  month: string;
  base_salary: number;
  bonus_amount?: number;
  penalty_amount?: number;
  kpi_percent?: number;
  kpi_rule_id?: number;
  comment?: string;
}

export interface PayrollUpdate {
  base_salary?: number;
  bonus_amount?: number;
  penalty_amount?: number;
  kpi_percent?: number;
  kpi_rule_id?: number;
  comment?: string;
}

export interface UpdatePayrollRequest {
  base_salary?: number;
  bonus_amount?: number;
  penalty_amount?: number;
  kpi_percent?: number;
  kpi_rule_id?: number;
  comment?: string;
}